### ISSUE TITLE
New layout for Go projects do not allow generate the CRDs with v1 version 

### ISSUE THREAD

**Issue by camilamacedo86** [operator-framework/operator-sdk#3580]
## Bug Report

**What did you do?**

$ mkdir $GOPATH/src/memcached-operator
$ cd $GOPATH/src/memcached-operator
$ operator-sdk init --domain example.com
$ operator-sdk create api  --group webapp --version v1 --kind Guestbook --crd-version=v1

**What did you expect to see?**
- Has the option to generate the crd-version with v1 or v1beta1 
- By default, gen CRD's and webhooks with v1

**What did you see instead? Under which circumstances?**

Unable to generate them using the v1 and the `v1beta1` will be deprecated soon for k8s. See the https://github.com/operator-framework/operator-sdk-samples/pull/137#discussion_r461361241 which motivates this issue as well. Also, it will not allow users that have been gen CRD's with `v1` via the flag migrate to the new layout without manual changes which shows a regression. 

**Environment**
* operator-sdk version: b95f052d11025bf6963215cbb9a331e5181b0c4f

**Possible solution**
- Wrapper Go plugin to allow use the --crd-version as it has for Helm/Ansible
- For CRD only: doc how to customize the Makefile to use `controller-tools crd:crdVersions=v1`. Update the makefile as:

```
# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
CRD_OPTIONS ?= "crd:crdVersions=v1" 
```

**Additional context**

In upstream we have the following issues to use v1 by default for v3-alpha. (https://github.com/kubernetes-sigs/kubebuilder/issues/1065, https://github.com/kubernetes-sigs/kubebuilder/issues/933) However, note that we cannot push it for v2 and for sdk shows that we also require to keep the possibility to generate the CRD's and WebHook manifests with both versions which are not required there since they can drop the supportability for old clusters.

**Comment by camilamacedo86** [comment]
c/c @joelanford @estroz @jmrodri

**Comment by estroz** [comment]
Generating v1 CRDs is a pretty straightforward change to the Makefile (https://github.com/kubernetes-sigs/kubebuilder/issues/1065#issuecomment-616823185), therefore this is an upstream issue.

**Comment by camilamacedo86** [comment]
Hi @estroz, 

The workaround solution to customize the makefile with `controller-tools crd:crdVersion=v1` can attend the CRD scenario but I am afraid that it cannot solve the Webhooks one. Also, by solving it in upstream we will not actually solve immediately here. since the change will be applied just for v3 and not necessary it will cover all our needs. See my comments in the section `Additional context`. Because of this, I raised this one for we are able to discuss.

**Comment by camilamacedo86** [comment]
Just to register:
- By updating the Makefile I could generate the CRD's v1 and all worked fine
- I could not gen webhooks manifest with v1 by updating the controller-gen
- OCP 4.4 shows support just v1: https://docs.openshift.com/container-platform/4.4/rest_api/extension_apis/validatingwebhookconfiguration-admissionregistration-k8s-io-v1.html

**Comment by joelanford** [comment]
@camilamacedo86 As @estroz mentioned, these issues need to be resolved upstream for the SDK to incorporate them. Can you create (if not already created) and link to the upstream issues where these problems are being discussed and fixed?

If they're already fixed and released upstream, we need an issue in the SDK to bump our dependencies.

**Comment by camilamacedo86** [comment]
Hi @joelanford, 

>. As @estroz mentioned, these issues need to be resolved upstream for the SDK to incorporate them Can you create (if not already created) and link to the upstream issues where these problems are being discussed and fixed?

See the body description of this issue. All links and content related upstream are described on it already. 
They was added when this issue was raised, see:  

<img width="1112" alt="Screen Shot 2020-09-21 at 15 53 53" src="https://user-images.githubusercontent.com/7708031/93782600-ac6af600-fc22-11ea-9700-84ead7d694ff.png">

> If they're already fixed and released upstream, we need an issue in the SDK to bump our dependencies.

It is not solved in upstream yet.  However, we will unable to bump the deps in this case. SDK will need to use v3+ to get this change from kb/upstream. We might allow users to work with the v3-alpha plugin but still using v2 as default to address it. WDYT?

**Comment by estroz** [comment]
All we need to do is import the v3-alpha kubebuilder plugin once https://github.com/kubernetes-sigs/kubebuilder/pull/1644 is merged.

**Comment by estroz** [comment]
Closed by #4282
